### PR TITLE
AzureDataShare: Adding version 2020-10-01-preview swagger 

### PR DESCRIPTION
**Pull request by ahermas**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ahermas** [comment]
@ChenTanyi I see that the PR has cli errors. I pinged the autorest team but I was wondering if you know about such errors and how they could be resolved. Thanks.

**Comment by ChenTanyi** [comment]
@ahermas The Cli CI has passed

**Comment by varunkumta** [comment]
>             "SystemAssigned"

is this a bug fix where the schema now reflects what the service supports? or is this an intentional functionality degradation from the previous api version?

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/DataShare.json:4710 in 0df549e. [](commit_id = 0df549e59d8fff1155181aae7800acbdf5a74c39, deletion_comment = False)

**Comment by varunkumta** [comment]
>         "containerName": {

consider modeling the containerName, location and storageAccountResourceId within a "storageAccount" object with those 3 fields

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/DataShare.json:6431 in 0df549e. [](commit_id = 0df549e59d8fff1155181aae7800acbdf5a74c39, deletion_comment = False)

**Comment by ahermas** [comment]
> > ```
> >         "SystemAssigned"
> > ```
> 
> is this a bug fix where the schema now reflects what the service supports? or is this an intentional functionality degradation from the previous api version?
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/DataShare.json:4710 in [0df549e](https://github.com/Azure/azure-rest-api-specs/commit/0df549e59d8fff1155181aae7800acbdf5a74c39). [](commit_id = [0df549e](https://github.com/Azure/azure-rest-api-specs/commit/0df549e59d8fff1155181aae7800acbdf5a74c39), deletion_comment = False)

@varunkumta  UserAssigned was added by mistake. My change is to remove it.

**Comment by varunkumta** [comment]
>         "location": {

as above, consider combining related storage account properties into 1 object

---
Refers to: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/DataShare.json:6492 in 0df549e. [](commit_id = 0df549e59d8fff1155181aae7800acbdf5a74c39, deletion_comment = False)

File: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/examples/DataSetMappings_StorageAccount_Create.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      },
+      "body": {
+        "kind": "StorageAccount",
+        "properties": {
+          "dataSetId": "a08f184b-0567-4b11-ba22-a1199336d226",
+          "storageAccountResourceId": "/subscriptions/433a8dfd-e5d5-4e77-ad86-90acdc75eb1a/resourceGroups/SampleResourceGroup/providers/Microsoft.Storage/storageaccounts/storageaccount1",
+          "containerName": "Container1",
+          "location": "centraluseuap",
+          "dataSetMappingStatus": "Ok",
+          "provisioningState": "Created"
```
Comment (varunkumta) [inline specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/examples/DataSetMappings_StorageAccount_Create.json:35]: >Created [](start = 32, length = 7)

this is not a valid enum value as per the schema, should be Succeeded

**Comment by sarajang** [comment]
> > ```
> >     "location": {
> > ```
> 
> as above, consider combining related storage account properties into 1 object
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/DataShare.json:6492 in [0df549e](https://github.com/Azure/azure-rest-api-specs/commit/0df549e59d8fff1155181aae7800acbdf5a74c39). [](commit_id = [0df549e](https://github.com/Azure/azure-rest-api-specs/commit/0df549e59d8fff1155181aae7800acbdf5a74c39), deletion_comment = False)

@varunkumta location is a read only property exposed to the customer. We would like to keep it a separate entry in the property bag.

**Comment by ahermas** [comment]
> > ```
> >     "containerName": {
> > ```
> 
> consider modeling the containerName, location and storageAccountResourceId within a "storageAccount" object with those 3 fields
> 
> Refers to: specification/datashare/resource-manager/Microsoft.DataShare/preview/2020-10-01-preview/DataShare.json:6431 in [0df549e](https://github.com/Azure/azure-rest-api-specs/commit/0df549e59d8fff1155181aae7800acbdf5a74c39). [](commit_id = [0df549e](https://github.com/Azure/azure-rest-api-specs/commit/0df549e59d8fff1155181aae7800acbdf5a74c39), deletion_comment = False)

@varunkumta  StorageAccountDataSetProperties already contains what you're referring to as a "storageaccount".

**Comment by ahermas** [comment]
@varunkumta @ChenTanyi @ArcturusZhang 
Santhosh and I have addressed/replied to your comments. Please let me know if you have further comments.
Thanks for reviewing.

**Review by ChenTanyi — APPROVED** [review]

**Comment by ChenTanyi** [comment]
@ahermas please let me know when it ready to merge.

**Comment by sarajang** [comment]
> @ahermas please let me know when it ready to merge.

@ChenTanyi  we found a minor issue and we are working on fixing it. Please hold merging this.

**Comment by ahermas** [comment]
> @ahermas please let me know when it ready to merge.

@ChenTanyi  The minor issue has been addressed.Please merge the PR when you get a chance.
Thanks

**Comment by ahermas** [comment]
Hi @ArcturusZhang,

I'm pinging you since I see the Chen will be away till the 17th.
Please kindly review the pr when you get a chance after the checks complete.

Thanks

**Pull request merged**